### PR TITLE
ECC private keys support for `tls` feature

### PR DESCRIPTION
**Pull request by joseluisq**
This PR adds support for [ECC private keys](https://cryptobook.nakov.com/asymmetric-key-ciphers/elliptic-curve-cryptography-ecc) when enabling the `tls` feature.
It is particularly useful when generating private keys using some client tools like for example [`Lego ACME`](https://github.com/go-acme/lego) which defaults to `EC256` keys.

Changelog:
- ECC private keys support
- Refactor `TlsConfigError` to reflect intended changes
- `tokio-rustls 0.24` and subsequent `client_auth` updates

### CODE REVIEW

File: src/tls.rs [hunk 1]
```diff
@@ -25,10 +25,12 @@ pub(crate) enum TlsConfigError {
     Io(io::Error),
     /// An Error parsing the Certificate
     CertParseError,
-    /// An Error parsing a Pkcs8 key
-    Pkcs8ParseError,
-    /// An Error parsing a Rsa key
-    RsaParseError,
```
Comment (joseluisq) [inline src/tls.rs:31]: I couldn't come up with any other idea than this removal. So please let me know your thoughts about it.

**Comment by mohe2015** [comment]
Is there anything for the author of this PR to do (maybe resolve conflicts) or would it be possible to maybe get this reviewed and merged?

**Comment by mohe2015** [comment]
Oh no please don't say this is a breaking change because of the dependency bump...

**Review by seanmonstar — APPROVED** [review]
This looks good to me, thanks!

**Comment by seanmonstar** [comment]
Looks like there's merge conflicts, if you resolve and ping me, I'll click merge :)

**Comment by joseluisq** [comment]
> Looks like there's merge conflicts, if you resolve and ping me, I'll click merge :)

Conflicts gone. Let me know if there is something else to do :).

**Review by seanmonstar — APPROVED** [review]

**Pull request merged**